File: src/hello_world_cpp_pkg/src/simple_sub.cpp
#include "rclcpp/rclcpp.hpp"

#include "std_msgs/msg/string.hpp"

class SimpleSubscriber : public rclcpp::Node
{
private:
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription;

    void topic_callback(const std_msgs::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "Received message: '%s'", msg->data.c_str());
    }

public:
    SimpleSubscriber() : Node("simple_subscriber")
    {
        subscription = this->create_subscription<std_msgs::msg::String>(
            "my_cool_topic", 10,
            std::bind(&SimpleSubscriber::topic_callback, this, std::placeholders::_1));
    }
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<SimpleSubscriber>();
    rclcpp::spin(node);

    rclcpp::shutdown();
    return 0;
}

File: src/hello_world_py_pkg/hello_world_py_pkg/simple_sub.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from std_msgs.msg import String

class SubscriberNode(Node):
    def __init__(self):
        super().__init__('simple_subscriber_node')
        
        self.subscription_ = self.create_subscription(
            String,
            'my_cool_topic',
            self.listener_callback,
            10  # Queue size
        )

    def listener_callback(self, msg: String):
        self.get_logger().info(f'Received message: {msg.data}')

def main(args=None):
    rclpy.init(args=args)

    node = SubscriberNode()
    rclpy.spin(node)

    rclpy.shutdown()

if __name__ == '__main__':
    main()